libcontainer/rootfs_linux.go

```diff
@@ -81,13 +86,22 @@ func prepareRootfs(pipe io.ReadWriter, iConfig *initConfig, mountFds mountFds) (
 		cgroupns:        config.Namespaces.Contains(configs.NEWCGROUP),
 	}
 	for i, m := range config.Mounts {
-		entry := mountEntry{Mount: m}
+		entry := mountEntry{Mount: m, idmapFD: -1}
 		// Just before the loop we checked that if not empty, len(mountFds) == len(config.Mounts).
 		// Therefore, we can access mountFds[i] without any concerns.
 		if mountFds.sourceFds != nil && mountFds.sourceFds[i] != -1 {
 			entry.srcFD = "/proc/self/fd/" + strconv.Itoa(mountFds.sourceFds[i])
 		}
 
+		// We validated before we can access idmapFds[i].
+		if mountFds.idmapFds != nil && mountFds.idmapFds[i] != -1 {
+			entry.idmapFD = mountFds.idmapFds[i]
+		}
+
+		if entry.idmapFD != -1 && entry.srcFD != "" {
+			return fmt.Errorf("malformed mountFds and idmapFds slice. Entry: %v has fds in both slices. One must be -1", i)
```

Comment:
I guess the "One must be -1" is not fully correct, because `srcFD` is expected to be either `/proc/self/fd/<FD>` or empty 😅

Reply:
haha, right! Great catch, I removed the last sentence and switched to use a comma before "Entry". Thanks!